src/qililab/result/qprogram/qblox_measurement_result.py

```diff
@@ -52,7 +53,8 @@ def array(self) -> np.ndarray:
         """
         path0 = self.raw_measurement_data["bins"]["integration"]["path0"]
         path1 = self.raw_measurement_data["bins"]["integration"]["path1"]
-        return np.array([path0, path1])
+
+        return np.array([path0, path1]) if self.shape is None else np.array([path0, path1]).reshape(*self.shape)
```

Comment:
If you reshape the whole array there should be a `2` in front for the I/Q: `reshape(2, *self.shape)`.

Reply:
Also I believe is more clear to do it in two steps. This will also trigger the codecov that tests are missing.

```suggestion
        array = np.array([path0, path1])
        if self.shape:
               array = array.reshape(2, *self.shape)
        return array
```